src/str.rs

```diff
@@ -448,3 +716,113 @@ impl<'ch> ParallelIterator for SplitWhitespace<'ch> {
             .drive_unindexed(consumer)
     }
 }
+
+
+// /////////////////////////////////////////////////////////////////////////
+
+/// Parallel iterator over substrings that match a pattern
+#[derive(Debug, Clone)]
+pub struct Matches<'ch, P: Pattern> {
+    chars: &'ch str,
+    pattern: P,
+}
+
+struct MatchesProducer<'ch, 'pat, P: Pattern + 'pat> {
+    chars: &'ch str,
+    pattern: &'pat P,
+}
+
+impl<'ch, P: Pattern> ParallelIterator for Matches<'ch, P> {
+    type Item = &'ch str;
+
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+        where C: UnindexedConsumer<Self::Item>
+    {
+        let producer = MatchesProducer { chars: self.chars, pattern: &self.pattern };
+        bridge_unindexed(producer, consumer)
+    }
+}
+
+impl<'ch, 'pat, P: Pattern> UnindexedProducer for MatchesProducer<'ch, 'pat, P> {
+    type Item = &'ch str;
+
+    fn split(self) -> (Self, Option<Self>) {
+        match split(self.chars) {
```

Comment:
So what am I missing here -- how do we know we are not splitting right in the middle of a match?

Reply:
Maybe this is just out of cache for me...

Reply:
At the moment, we only support single-character patterns for splits and matches.